### PR TITLE
Add fabric files targeting composition

### PR DESCRIPTION
**Pull request by acoates-ms**
This adds the files necessary for a fabric implementation using composition.  These changes are limited to the fabric code path which is disabled by default.

I don't expect a complete review of these files to be done at this time -- this code is still a work in progress. This mostly gets the files into the repo, so that continuing work and progress can be reviewed and tracked properly.

This does not include a playground app that can run this - which will come in a subsequent change.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/10594)

### CODE REVIEW

File: vnext/Microsoft.ReactNative/CompositionContext.idl [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    DOC_STRING(
+      "Creates a @IVisual from a Windows.UI.Composition.Visual")
+    static IVisual CreateVisual(Windows.UI.Composition.Visual visual);
+
+    static Windows.UI.Composition.Compositor InnerCompositor(ICompositionContext context);
+    static Windows.UI.Composition.Visual InnerVisual(IVisual visual);
+    static Windows.UI.Composition.DropShadow InnerDropShadow(IDropShadow shadow);
+    static Windows.UI.Composition.CompositionBrush InnerBrush(IBrush brush);
+    static Windows.UI.Composition.ICompositionSurface InnerSurface(ICompositionDrawingSurface surface);
```
Comment (asklar) [inline vnext/Microsoft.ReactNative/CompositionContext.idl:28]: I think we will need a similar trick as we use for xaml to easily target lifted comp, so COMP_NAMESPACE that switches from Windows.UI.Composition to Microsoft.UI.Composition

File: vnext/Microsoft.ReactNative/CompositionSwitcher.idl [hunk 2]
```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "NamespaceRedirect.h"
+#include "DocString.h"
+
+namespace Microsoft.ReactNative.Composition
```
[the remark this answers is not in the document]
Comment (acoates-ms) [reply vnext/Microsoft.ReactNative/CompositionSwitcher.idl:7]: The composition files are not included in the winmd currently, as they are only included if the UseFabric build flag is set.

File: vnext/Microsoft.ReactNative/CompositionContext.idl (hunk 1 above)
Comment (acoates-ms) [reply vnext/Microsoft.ReactNative/CompositionContext.idl]: Done

**Review by vmoroz — APPROVED** [review]
:shipit:

**Pull request merged**